### PR TITLE
[PREVIEW] - redirect if error on CYA

### PR DESCRIPTION
**Pull request by dtaylor7**
# Description

[DIV-3428 - Petitioner - DN application - error on CYA if can't update CCD](https://tools.hmcts.net/jira/browse/DIV-3428)

If there is an error when submitting an application to CCD show the CYA page and display an error

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "@hmcts/look-and-feel": "^1.13.2",
     "@hmcts/nodejs-healthcheck": "^1.4.6",
     "@hmcts/nodejs-logging": "https://github.com/hmcts/nodejs-logging#771bb219e86d79d999d7edb17b10c69194d614fd",
-    "@hmcts/one-per-page": "^4.0.3",
+    "@hmcts/one-per-page": "https://github.com/hmcts/one-per-page.git#af675c8cc53cc19316c44de9c1bc67789fa48919",
```
Comment (dtaylor7) [inline package.json]: not fetching from git for some reason

File: package.json [hunk 2]
```diff
@@ -59,7 +59,7 @@
   },
   "devDependencies": {
     "@hmcts/div-idam-test-harness": "https://github.com/hmcts/div-idam-test-harness.git",
-    "@hmcts/one-per-page-test-suite": "^2.0.10",
+    "@hmcts/one-per-page-test-suite": "https://github.com/hmcts/one-per-page-test-suite.git#95a9c95b5e202c3a33ce4b2fd563f453e3439fde",
```
Comment (dtaylor7) [inline package.json]: not fetching from git for some reason

File: steps/check-your-answers/CheckYourAnswers.step.js [hunk 3]
```diff
@@ -18,13 +18,23 @@ class CheckYourAnswers extends CYA {
   next() {
     return action(caseOrchestrationService.submitApplication)
       .then(goTo(this.journey.steps.Done))
-      .onFailure(redirectTo(this.journey.steps.CheckYourAnswers));
+      .onFailure((error, req, res) => {
+        const { session } = req;
+        // push error into session to display error message to user
+        session.temp = { CheckYourAnswers: Object.assign({}, session.CheckYourAnswers, { submitError: 'error' }) };
```
Comment (qzhou-hmcts) [inline steps/check-your-answers/CheckYourAnswers.step.js]: Could you also log the error here?

Comment (dtaylor7) [reply steps/check-your-answers/CheckYourAnswers.step.js]: it is already logged in two places, One per page and Case Orchestration Service

**Pull request merged**